tests/keras/backend/backend_test.py

```diff
@@ -599,6 +600,93 @@ def rnn_fn(x_k, h_k):
                 assert_allclose(outputs_list[i - 1], outputs_list[i], atol=1e-05)
                 assert_allclose(state_list[i - 1], state_list[i], atol=1e-05)
 
+    def test_rnn_additional_states(self):
+        # implement a simple RNN with an additional state
+        # whose shape is different from that of the output
+        num_samples = 4
+        input_dim = 5
+        output_dim = 3
+        state_dim = 7
+        timesteps = 6
+
+        _, x = parse_shape_or_val((num_samples, timesteps, input_dim))
+        _, h0 = parse_shape_or_val((num_samples, output_dim))
+        _, s0 = parse_shape_or_val((num_samples, state_dim))
+        _, wi = parse_shape_or_val((input_dim, output_dim))
+        _, wh = parse_shape_or_val((output_dim, output_dim))
+        _, ws = parse_shape_or_val((state_dim, output_dim))
+        _, wxs = parse_shape_or_val((input_dim, state_dim))
+        _, wss = parse_shape_or_val((state_dim, state_dim))
+        mask = np.random.randint(2, size=(num_samples, timesteps))
+
+        x_k = K.variable(x)
+        hs0_k = [K.variable(h0), K.variable(s0)]
+        wi_k = K.variable(wi)
+        wh_k = K.variable(wh)
+        ws_k = K.variable(ws)
+        wxs_k = K.variable(wxs)
+        wss_k = K.variable(wss)
+        mask_k = K.variable(mask)
+
+        def rnn_fn(x_k, hs_k):
```

Comment:
Does it really have to be this complex? This is just a unit test.

why not:

```python
def rnn_fn(x, h):
   return x, [x, K.concatenate([x, x], axis=-1)]
```

Reply:
Thanks for your suggestions! I simplified the unit_test.